crafters/nlp/TikaExtractor/__init__.py

```diff
@@ -0,0 +1,60 @@
+__copyright__ = "Copyright (c) 2020 Jina AI Limited. All rights reserved."
+__license__ = "Apache-2.0"
+
+import os
+
+import requests
+from requests.exceptions import ConnectionError
+from jina.executors.crafters import BaseCrafter
+
+TIKA_URL = 'http://0.0.0.0:9998'
+
+
+class TikaExtractor(BaseCrafter):
+    """
+    :class:`TikaExtractor` Extracts text from files.
+    """
+
+    def start_tika(self):
+        from subprocess import Popen
+        tika_version = os.getenv('TIKA_VERSION')
+        _ = Popen(
+            ['java', '-jar', f'/tika-server-{tika_version}.jar', '-h', '0.0.0.0'],
+            stdout=open('stdout.log', 'w'),
+            stderr=open('stderr.log', 'w'))
+
+        response = None
+        while not response:
+            try:
+                response = requests.get(TIKA_URL)
+            except ConnectionError:
+                pass
+
+    def __init__(self, *args, **kwargs):
+        self.start_tika()
+        super().__init__(*args, **kwargs)
+
+    def craft(self, uri: str, *args, **kwargs):
```

Comment:
Is there a chance for it to accept bytes? And can it return images as well as text, if so it would be cool to have it as a Segmenter

Reply:
Done.

Reply:
Hi cool! just wanted to ask the same, can it work with images too?

Reply:
Yes, Tika will apply tesseract ocr internally AFAIK.

Reply:
Nice! I asked cause I tried it too but wasn't working very well for me in the tests, but cool you made it work!
But then if it returns both I think it should be on the Segmenter, since Crafters are supposed to work with 1-1 documents

Reply:
Yes, is there a possibility that `tika` returns more than one `result` in output? Like an output for text and some other output for images? or how does it behave on that?

Reply:
Tika would return text anyway. It would not return the detected images. So the input could be images, pdfs, docx, ... but output would be always text. In this case it should be a crafter?

Reply:
Yes, in this case it should be a crafter yes! Then in any downstream segmenter u could split all the text as u want.